packages/backend/src/ee/services/ai/agents/tests/agent.test.ts

```diff
@@ -0,0 +1,190 @@
+import { CoreAssistantMessage, CoreMessage } from 'ai';
+import fetchMock from 'jest-fetch-mock';
+import { generateAgentResponse } from '../agent';
+import {
+    createArgs as createArgsFactory,
+    createMessage,
+    createMockDepsFactory,
+    createToolMessage,
+    createToolMessages,
+    mockExploresSummary,
+    promptTestUtils,
+} from './testUtils';
+
+const TIMEOUT = 30_000;
+const createArgs = createArgsFactory({
+    apiKey: process.env.OPENAI_API_KEY!,
+    modelName: 'gpt-4.1',
+});
+
+describe('generateAgentResponse', () => {
+    beforeEach(() => {
+        fetchMock.disableMocks();
+        jest.clearAllMocks();
+    });
+
+    describe('Basic functionality', () => {
+        it(
+            'correctly returns all the models it can use',
+            async () => {
+                const message = createMessage('What models can you use?');
+                const args = createArgs({
+                    messageHistory: [message],
+                    agentSettings: {
+                        instruction: `When asked about models you can use, only respond with a list of the models you can see`,
```

Comment:
I am curious to remove all special instructions and see how agent performs. we'll never have that kind of instructions in real scenarios.